README.md

```diff
@@ -171,8 +177,82 @@ datalad containers-run \
         --output . \
         '{inputs}' '{outputs}' participant group -w workdir
 )
+
+```
+</details>
+
+## Walkthrough
+
+For users who are new to these components, we will walk through how
+these components are used together in a typical Yoda workflow.
+the steps
+
+```bash
+mkdir ~/my-experiments
+cd ~/my-experiments
+datalad create -d ds000003-qc -c text2git
+cd ds000003-qc
+```
+
+Datalad has created a new directory for our results, `ds000003-qc`.
+According to YODA principles, this dataset should also contain our input
+data, code, and anything else we need to run the analysis.
+
+Install the input dataset:
+
+```bash
+datalad install -d . -s https://github.com/ReproNim/ds000003-demo sourcedata
+```
+
+Next we install the `ReproNim/containers` collection.
+
+```bash
+datalad install -d . ///repronim/containers
 ```
 
+Now let's take a look at what we have.
+
+```bash
+/ds000003-qc # The root dataset contains everything
+    /sourcedata # we call it source, but it is actually 000003-demo
+    /repronim/containers # This is where our non-custom code lives
+```
+
+Now we can "freeze" a container or containers that we plan to use so
+anyone (including you) will know exactly which version was used. This is
+technically optional, but very helpful. (All that changes is a
+subproject commit hash.)
+
+```bash
+datalad run -m "Downgrade/Freeze mriqc container version" \
+    containers/scripts/freeze_versions bids-mriqc=0.16.0
+```
+
+When we run the bids-mriqc container, it will need a working directory
+for intermediate files. These are not helpfull to commit, so we will
+tell git (and datalad) to ignore the whole directory.
+
+```bash
+echo "workdir/" > .gitignore && datalad save -m "Ignore workdir" .gitignore
+```
+
+Now we use `datalad containers-run` to perform the analysis.
+
+**_NOTE:_** mriqc is memory intensive, so we are restricting to a single
+participant. OOM error will cause a NodeExecutionError, and the cause
+may not be immediately obvious.
+
+```bash
+datalad containers-run \
+        --container-name containers/bids-mriqc \
+        --input sourcedata \
+        --output . \
+        '{inputs}' '{outputs}' participant --participant-label sub-02 -w workdir
+```
+
+If everything worked as expected, we will now see our new analysis, and
+a commit message of how it was obtained!
+
 and now you have a dataset which has a git record on how these data
```

Comment:
join into above or just make it capitalized

Reply:
joined in and rewrote.